src/domain/usecases/SyncMetadataUseCase.ts

```diff
@@ -0,0 +1,259 @@
+import _ from "lodash";
+
+import { Maybe } from "utils/ts-utils";
+import { Async } from "domain/entities/Async";
+import { isUserModel, MetadataModel, MetadataObject } from "domain/entities/MetadataObject";
+import { MetadataRepository } from "domain/repositories/MetadataRepository";
+import {
+    ExclusiveMetadataItem,
+    MetadataValidationResult,
+    DiscrepancyMetadata,
+    DiscrepancyValidationResult,
+    METADATA_PROPERTIES_TO_IGNORE,
+} from "domain/entities/MetadataValidationResult";
+import logger from "utils/log";
+
+export class SyncMetadataUseCase {
+    constructor(
+        private metadataRepositoryMain: MetadataRepository,
+        private metadataReplicaRepositories: MetadataRepository[]
```

Comment:
Not sure what would be the best approach. As we have multiple instances to target as part of the domain app, maybe NOW it will make sense to have the concept of "Instance". This way, we only have a single MetadataRepository.

Reply:
After discussing this with Jorge he told me that since the Instance information is only needed to fetch data, but without no impact on the Metadata logic I can keep the same implementation.

I've created an Instance entity anyway, but just in case we need logic around it in the future. That way we avoid adding more code in the presentation layer.